src/Illuminate/Console/Scheduling/Schedule.php

```diff
@@ -97,6 +97,7 @@ public function job($job, $queue = null, $connection = null)
             $job = is_string($job) ? resolve($job) : $job;
 
             if ($job instanceof ShouldQueue) {
+                $queue = $queue ?? $job->queue;
```

Comment:
Any reason why you don't also consider the job connection?

Reply:
I just resubmitted #25159 so that it gets merged ASAP before 5.7 gets released which is basically any day now, so I didn't even consider the connection. But you are right, the connection should also be respected so I updated the PR accordingly.